nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/trash/AbstractTrashService.java

```diff
@@ -303,34 +304,49 @@ public void untrashDocuments(List<DocumentModel> docs) {
      */
     protected static final Pattern COLLISION_PATTERN = Pattern.compile("(.*)\\.[0-9]{13,}");
 
+    @Override
+    public boolean isMangledName(String docName) {
+        return TRASHED_PATTERN.matcher(docName).matches();
+    }
+
     @Override
     public String mangleName(DocumentModel doc) {
         return doc.getName() + "._" + System.currentTimeMillis() + "_.trashed";
     }
 
     @Override
     public String unmangleName(DocumentModel doc) {
-        String name = doc.getName();
-        Matcher matcher = TRASHED_PATTERN.matcher(name);
-        if (matcher.matches() && matcher.group(1).length() > 0) {
-            name = matcher.group(1);
-            matcher = COLLISION_PATTERN.matcher(name);
-            if (matcher.matches() && matcher.group(1).length() > 0) {
-                CoreSession session = doc.getCoreSession();
-                if (session != null) {
-                    String orig = matcher.group(1);
-                    String parentPath = session.getDocument(doc.getParentRef()).getPathAsString();
-                    if (parentPath.equals("/")) {
-                        parentPath = ""; // root
-                    }
-                    String newPath = parentPath + "/" + orig;
-                    if (!session.exists(new PathRef(newPath))) {
-                        name = orig;
-                    }
-                }
+        return unmangleName(doc.getCoreSession(), doc.getName(), doc.getParentRef());
+    }
+
+    @Override
+    public String unmangleName(CoreSession session, String docName, DocumentRef parentRef) {
+        if (session == null) {
+            return docName;
+        }
+        Matcher matcher = TRASHED_PATTERN.matcher(docName);
+        if (!matcher.matches()) {
+            return docName;
+        }
+        if (StringUtils.isEmpty(matcher.group(1))) {
+            return docName;
+        }
+        docName = matcher.group(1);
+        matcher = COLLISION_PATTERN.matcher(docName);
+        if (!matcher.matches()) {
+            return docName;
+        }
+        if (StringUtils.isNotEmpty(matcher.group(1))) {
+            String orig = matcher.group(1);
+            String parentPath = session.getDocument(parentRef).getPathAsString();
+            if (parentPath.equals("/")) {
+                parentPath = ""; // root
+            }
+            String newPath = parentPath + "/" + orig;
+            if (!session.exists(new PathRef(newPath))) {
+                docName = orig;
```

Comment:
I would have kept the exact same code as in the old `public String unmangleName(DocumentModel doc)` method here, to have the smallest diff possible on 10.10.